crates/typst-library/src/model/outline.rs

```diff
@@ -59,16 +70,89 @@ use crate::text::{LinebreakElem, LocalName, SpaceElem, TextElem};
+/// it and its entries. This way, you can customize the outline's
+/// [`title`]($outline.title), how outline entries are
+/// [indented]($outline.indent), and how the space between an entry's text and
+/// its page number should be [filled]($outline.entry.fill).
+///
+/// Richer customization is possible through configuration of the outline's
+/// [entries]($outline.entry). The outline generates one entry for each outlined
+/// element.
+///
+/// ## Spacing the entries { #entry-spacing }
+/// Outline entries are [blocks]($block), so you can adjust the spacing between
+/// them with normal block-spacing rules:
+///
+/// ```example
+/// #show outline.entry.where(
+///   level: 1
+/// ): set block(above: 1.2em)
+///
+/// #outline()
+///
+/// = About ACME Corp.
+/// == History
+/// === Origins
+/// = Products
+/// == ACME Tools
+/// ```
+///
+/// ## Building an outline entry from its parts { #building-an-entry }
+/// For full control, you can also write a transformational show rule on
+/// `outline.entry`. However, the logic for properly formatting and indenting
+/// outline entries is quite complex and the outline entry itself only contains
+/// two fields: The level and the outlined element.
+///
+/// For this reason, various helper functions are provided. You can mix and
+/// match these to compose an entry from just the parts you like.
+///
+/// The default show rule for an outline entry looks like this[^1]:
+/// ```typ
+/// #show outline.entry: it => link(
+///   it.element.location(),
+///   it.indented(it.prefix(), it.inner()),
+/// )
+/// ```
+///
+/// - The [`indented`]($outline.entry.indented) function takes an optional
+///   prefix and inner content and automatically applies the proper indentation
+///   to it, such that different entries align nicely and long headings wrap
+///   properly.
+///
+/// - The [`prefix`]($outline.entry.prefix) function formats the element's
+///   numbering (if any). It also appends a supplement for certain elements.
+///
+/// - The [`inner`]($outline.entry.inner) function combines the element's
+///   [`body`]($outline.entry.body), the filler, and the
+///   [`page` number]($outline.entry.page).
+///
+/// You can use these individual functions to format the outline entry in
+/// different ways. Let's say, you'd like to fully remove the filler and page
+/// numbers. To achieve this, you could write a show rule like this:
+///
+/// ```example
+/// #show outline.entry: it => link(
+///   it.element.location(),
+///   // Keep just the body, dropping
+///   // the fill and the page.
+///   it.indented(it.prefix(), it.body()),
+/// )
+///
+/// #outline()
+///
+/// = About ACME Corp.
+/// == History
+/// ```
+///
+/// [^1]: The outline of equations is the exception to this rule as it does not
+///       have a body and thus does not use indented layout.
+#[elem(scope, keywords = ["Table of Contents"], Show, ShowSet, LocalName, Locatable)]
```

Comment:
```suggestion
#[elem(scope, keywords = ["Table of Contents", "toc"], Show, ShowSet, LocalName, Locatable)]
```

"toc" is a common abbreviation of "table of contents" people might search for.

Reply:
I added "toc" but left "index" out because index is what's typically at the back of a book.